app/updatemanager.go

```diff
@@ -281,6 +289,10 @@ func (m UpdateControlMap) Sanitize() {
 	}
 }
 
+func (u *UpdateControlMap) Expired() bool {
+	return time.Now().After(u.expiry)
+}
+
 type ControlMap struct {
```

Comment:
Reading the code now, I think this should be `ControlMapPool` or something like that. Methods like `Delete` or `Contains` would read better.

Reply:
Yes, agree